src/samd/Servo.cpp

```diff
@@ -93,6 +93,13 @@ void Servo_Handler(timer16_Sequence_t timer, Tc *tc, uint8_t channel, uint8_t in
         }
 
         // Get the counter value
+#if defined(__SAMD51__)
+        // Note from datasheet: Prior to any read access, this register must be synchronized by user by writing the according TC
+        // Command value to the Control B Set register (CTRLBSET.CMD=READSYNC)
+        while (tc->COUNT16.SYNCBUSY.bit.CTRLB);
+	tc->COUNT16.CTRLBSET.bit.CMD = TC_CTRLBSET_CMD_READSYNC_Val;
+        while (tc->COUNT16.SYNCBUSY.bit.CTRLB);
```

Comment:
Not sure but in Marlin I did something like:
`while(tc->COUNT16.SYNCBUSY.bit.CTRLB || tc->COUNT16.SYNCBUSY.bit.COUNT)`

Reply:
i think this is equivalent, we sync after the read (but not sure its needed?)

Reply:
I also sync with count but as told in other thread, I don't have a valid shield to test this (and other) AGCM4 code...sigh

Reply:
ok we are using an oscilloscope to verify functionality.

Reply:
How lucky you are....;-D